src/core/algorithms/dc/parser/dc_parser.cpp

```diff
@@ -0,0 +1,174 @@
+
+    dc_string_.erase(dc_string_.begin());
+    dc_string_.erase(std::prev(dc_string_.end()));
+
+    if (dc_string_.empty()) throw std::invalid_argument("Empty DC is not allowed");
+
+    std::vector<Predicate> res;
+    while (has_next_predicate_) {
+        res.push_back(GetNextPredicate());
+    }
+
+    return {std::move(res)};
+}
+
+Predicate DCParser::GetNextPredicate() {
+    size_t ind = dc_string_.find(kSep, cur_);
+    if (ind == std::string::npos) has_next_predicate_ = false;
+
+    std::string str_pred = dc_string_.substr(cur_, ind - cur_);
+    cur_ = ind + kSep.size();
+
+    return ConvertToPredicate(std::move(str_pred));
+}
+
+Predicate DCParser::ConvertToPredicate(std::string const& pred) {
+    size_t ind;
+    auto check = [&ind, &pred](std::string const& op) {
+        std::string spaced_op = ' ' + op + ' ';
+        ind = pred.find(spaced_op);
+        return ind != std::string::npos;
+    };
+
+    auto it = std::ranges::find_if(str_operators_, check);
+    if (it == str_operators_.end()) throw std::invalid_argument("Missing permissible operator");
+
+    std::string str_operator = *it;
+    size_t right_start = ind + str_operator.size() + 1;
+    auto str_left_operand = pred.substr(0, ind);
+    auto str_right_operand = pred.substr(right_start);
+
+    boost::trim(str_left_operand);
+    boost::trim(str_right_operand);
+
+    auto [left_operand, right_operand] = GetOperands(str_left_operand, str_right_operand);
+
+    return {std::move(str_operator), std::move(left_operand), std::move(right_operand)};
+}
+
+std::pair<ColumnOperand, ColumnOperand> DCParser::GetOperands(std::string str_left_op,
+                                                              std::string str_right_op) {
+    std::optional<ColumnOperand> left_op, right_op;
+
+    if (IsVarOperand(str_left_op)) {
+        RemoveEscaping(str_left_op);
+        left_op = ConvertToVariableOperand(str_left_op);
+    }
+
+    if (IsVarOperand(str_right_op)) {
+        RemoveEscaping(str_right_op);
+        right_op = ConvertToVariableOperand(str_right_op);
+    }
+
+    if (!left_op.has_value() and !right_op.has_value())
+        throw std::invalid_argument("Pure constant predicate is not allowed");
+
+    if (!left_op.has_value()) {
+        left_op = GetOperand(*right_op, str_left_op);
+    } else if (!right_op.has_value()) {
+        right_op = GetOperand(*left_op, str_right_op);
+    }
+
+    return {std::move(*left_op), std::move(*right_op)};
+}
+
+void DCParser::RemoveEscaping(std::string& str) const {
+    size_t ind = str.find_first_of('\\');
+    if (ind != std::string::npos) str.erase(ind, 1);
+}
+
+bool DCParser::IsVarOperand(std::string op) const {
+    return op.starts_with("t.") or op.starts_with("s.");
+}
+
+ColumnOperand DCParser::ConvertToVariableOperand(std::string const& operand) const {
+    dc::Tuple tuple = operand.front() == 't' ? dc::Tuple::kT : dc::Tuple::kS;
+    std::string name = operand.substr(2);
```

Comment:
I think we can make this a `string_view` to avoid extra copies
```suggestion
    std::string_view name = operand.substr(2);
```